### PR TITLE
Allow passing `env` to render tests

### PR DESCRIPTION
**Pull request by sadlerjw**
This allows me to add tests for the suggested implementation on https://github.com/markdown-it/markdown-it-footnote/pull/7

As usual, happy to be sent back to the drawing board with better ideas :)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ var p       = require('path');
 
 var _       = require('lodash');
 var yaml    = require('js-yaml');
+var merge   = require('merge');
```
Comment (puzrin) [inline index.js]: No microdeps please, `lodash` already has `assign()` and other similar functions.

File: index.js [hunk 2]
```diff
@@ -168,10 +169,15 @@ function load(path, options, iterator) {
 }
 
 
-function generate(path, options, md) {
+function generate(path, options, env, md) {
```
Comment (puzrin) [inline index.js]: 1. I suspect that will break existing API.
2. If you passed `options` OR `env` (only one), that will fail to detect param type

Why not add `env` as last param?

**Comment by puzrin** [comment]
Hm... by the way... do you really need this change? There are `.load()` method to use custom test wrappers.

**Comment by sadlerjw** [comment]
Good question - if core markdown-it eventually gets similar functionality, through similar mechanisms (as discussed on the footnote PR), presumably passing in `env` will become a more-frequently-used use case. If you disagree, I'm happy to abandon this PR and basically inline the `generate` implementation into my test script in the footnote PR.

Either way, I've addressed your other concerns.

**Comment by puzrin** [comment]
My approach is to not extend api "in advance". Usually API is extented when multiple real examples available (single is not enougth) or if someone is 200% sure that it's needed.

I'd suggest to close PR now and retun back after plugin finished (if that's still required).

**Comment by sadlerjw** [comment]
:thumbsup:

**Pull request closed without merging**